### PR TITLE
Use private temporary folder for performance tests

### PR DESCRIPTION
**Pull request by tworthington**
We felt that even though the write and read to the temporary folder is currently quite trivial, it should still go into its own subfolder so this feature branch adds report_benchmark to the CFG->tempdir path and the plugin then writes to that.

### CODE REVIEW

**Comment by martignoni** [comment]
Good idea!

Can you elaborate about the "why"? is it for safety reasons?

**Comment by tworthington** [comment]
> Good idea!
> 
> Can you elaborate about the "why"? is it for safety reasons?

It's partly just neatness but it's also connected to how the temp folder is used. Course backups are constructed there and then moved into the coursebackup area. Because course backups can be tens of gigs in size, they can't go into a ramdisc so we end up assigning some of temp to ramdisc and some to hard-drive or SSD. If each plugin that uses temp has its own folder then we can have fine control of which parts go into ram and which don't.

Basically, we assume that anything that writes into temp will require gigs of space until proven otherwise. Once we're happy that it doesn't (sessions, for example), it can be sped up by mounting a ramdisc partition over its specific subdirectory.

If benchmark writes directly to temp we can't easily do this and it becomes an all-or-nothing issue when we are measuring performance changes and that in turn risks problems with course backups if someone forgets to switch things back after the testing.

**Comment by Lenacus** [comment]
Look the https://github.com/mikasmart/moodle-report_benchmark/issues/10

What do you think is best for defining the temporary file?

**Comment by martignoni** [comment]
@tworthington Thanks for the explanation.

It would then be useful to have this every time we use the temp folder. Would you agree to do the same modification for all tests? And, as @Lenacus writes, what about implementing this with `make_request_directory`, as in https://github.com/mikasmart/moodle-report_benchmark/issues/10#issuecomment-236470439?

**Comment by tworthington** [comment]
@martignoni I'll have a tinker with `make_request_directory` and report back; I've not used it before. I'm happy to apply the same method to any other tests that would benefit.

**Comment by martignoni** [comment]
In fact using `make_request_directory()` doesn't use temporary folder. Re-reading [cited comment](https://github.com/mikasmart/moodle-report_benchmark/issues/10#issuecomment-236470439), we should instead use `make_temp_directory ()`.

`make_request_directory()` will be useful to build tests on the localcachedir in addition to the tempdir.

**Comment by martignoni** [comment]
Commit 5af18c90945f55cf3742acc5b6e1799cdc948c14 obsoleted this PR.

@tworthington Thanks anyway for the good idea and the work done.

**Comment by tworthington** [comment]
> Commit [5af18c9](https://github.com/mikasmart/moodle-report_benchmark/commit/5af18c90945f55cf3742acc5b6e1799cdc948c14) obsoleted this PR.
> 
> @tworthington Thanks anyway for the good idea and the work done.

No problem. I'd just logged into say that make_request_directory() doesn't use temporary folder.
